tests/PHPStan/Type/TypeCombinatorTest.php

```diff
@@ -4134,6 +4162,19 @@ public function dataIntersect(): iterable
 			IntersectionType::class,
 			'array{a?: true, c?: true}&non-empty-array',
 		];
+		yield [
+			[
+				new ConstantStringType(Exception::class),
+				new GenericClassStringType(TemplateTypeFactory::create(
+					TemplateTypeScope::createWithFunction('foo'),
+					'T',
+					null,
+					TemplateTypeVariance::createInvariant(),
+				)),
+			],
+			ConstantStringType::class,
+			"'Exception'",
```

Comment:
Please test intersection with `'array'` too - should be NeverType.

Reply:
Fixing the test for phpstan/phpstan#6697 would make it `'array'&class-string<T (function foo(), parameter)>` instead of `never`, because we would need to assume that `'array'` may be name of a class PHPStan doesn't know about. What should I do about that?